### ISSUE TITLE
Dropout inside GATConv layer

### ISSUE THREAD

**Issue by afternone** [CarloLucibello/GraphNeuralNetworks.jl#258]
In the [(Optional) Exercises](https://carlolucibello.github.io/GraphNeuralNetworks.jl/stable/tutorials/introductory_tutorials/node_classification_pluto/#(Optional)-Exercises), it is mentioned dropout inside GATConv layer, but I did not find the keywords parameter about dropout. How can I set dropout inside GATConv layer.

**Comment by CarloLucibello** [comment]
Actually, that is a missing feature. We should add the `dropout` option to `GATConv` and `GATv2Conv` 
PyG [has it](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GATConv.html)

**Comment by 5hv5hvnk** [comment]
@CarloLucibello 
Can adding something like this after [line 366](https://github.com/CarloLucibello/GraphNeuralNetworks.jl/blob/9aa64ffc8f652d382d09496e7993a34aefe63d6f/src/layers/conv.jl#L366) here can do the job?
```
if l.dropout > 0 && Flux.training()
        x = Flux.dropout(x, l.dropout)
```

**Comment by CarloLucibello** [comment]
not really, dropout is performed on the attention coefficients I think. 
It corresponds to masking the attention, see https://github.com/FluxML/NNlib.jl/blob/master/src/attention.jl
